lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -3066,6 +3066,12 @@ bool LowererMDArch::GenerateFastDivAndRem(IR::Instr* instrDiv, IR::LabelInstr* b
     IR::Opnd* divisor   = instrDiv->GetSrc2(); // denominator
     IR::Opnd* dst       = instrDiv->GetDst();
 
+    if(static_cast<IR::RegOpnd*>(divident)->IsSameRegUntyped(dst) && (instrDiv->m_opcode == Js::OpCode::Rem_I4 || instrDiv->m_opcode == Js::OpCode::RemU_I4 || bailOutLabel))
```

Comment:
seems a bit unsafe to do static cast here. i'd prefer if we had code like `if(!divident->IsRegOpnd()) return false;` to guard the cast

Reply:
Yes, will do. Thanks

Reply:
Also, use `dividend->AsRegOpnd()` instead of a cast

Reply:
Yes already did that :) no more static cast..